autobahn/wamp/component.py

```diff
@@ -343,6 +344,20 @@ def describe_endpoint(self):
         return repr(self.endpoint)
 
 
+def type_check(func):
+    async def _type_check(*args, **kwargs):
+        # Converge both args and kwargs into a dictionary
+        arguments = inspect.getcallargs(func, *args, **kwargs)
+        response = []
+        for name, kind in func.__annotations__.items():
+            if name in arguments and type(arguments[name]) != kind:
+                response.append("'{}' required={} got={}".format(name, type(arguments[name]).__name__, kind.__name__))
```

Comment:
Should this test be `isinstance` instead? (I think it should; if I demand `Base` as the type, I'd expect my subclass to be accepted)

Reply:
done